src/main/java/hudson/plugins/s3/S3BucketPublisher.java

```diff
@@ -206,11 +214,13 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launc
 
         if (profile == null) {
             log(Level.SEVERE, console, "No S3 profile is configured.");
-            run.setResult(constrainResult(Result.UNSTABLE, listener));
-            return;
+            if (!isDontSetBuildResultOnFailure()) {
+                run.setResult(constrainResult(Result.UNSTABLE, listener));
+                return;
+            }
+            throw new IOException("No S3 profile is configured.");
```

Comment:
Use `hudson.AbortException` instead as displaying a stacktrace would not serve any purpose here.